sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/JacksonTypeScanner.java

```diff
@@ -0,0 +1,113 @@
+        .collect(Collectors.joining(System.lineSeparator()));
+  }
+
+  private List<MappableField> scan(List<String> rootPath, JavaType type) {
+    try {
+      BeanDescription introspect = mapper.getSerializationConfig().introspect(type);
+      List<MappableField> fields = new ArrayList<>();
+      for (var property : introspect.findProperties()) {
+        String name = property.getName();
+        var newRootPath = new ArrayList<>(rootPath);
+        newRootPath.add(name);
+        JavaType propertyType = getDeclaredType(property.getPrimaryType());
+        if (propertyType == null) {
+          LOG.info("Could not find type of field at path {}", newRootPath);
+          continue;
+        }
+        if (propertyType.isPrimitive()
+            || propertyType.hasRawClass(String.class)
+            || propertyType.isEnumType()
+            || propertyType.isEnumImplType()
+            || propertyType.hasRawClass(File.class)) {
+          fields.add(new MappableField(newRootPath, propertyType.getRawClass()));
+        } else if (propertyType.isArrayType()) {
+          LOG.info("Found array type for {}. Arrays are not supported yet.", newRootPath);
+        } else if (Map.class.isAssignableFrom(propertyType.getRawClass())) {
+          fields.add(new MappableField(newRootPath, Map.class));
+        } else {
+          fields.addAll(scan(newRootPath, propertyType));
+        }
+      }
+      return fields;
+    } catch (Exception e) {
+      LOG.info("Could not parse config class {} for dynamic environment properties.", type, e);
+      return List.of();
+    }
+  }
+
+  private JavaType getDeclaredType(JavaType primaryType) {
+    if (Optional.class.isAssignableFrom(primaryType.getRawClass())) {
+      return getDeclaredType(primaryType.findTypeParameters(Optional.class)[0]);
+    }
+    if (primaryType.getRawClass().isAnnotationPresent(JsonTypeInfo.class)) {
+      var jsonTypeInfo = primaryType.getRawClass().getAnnotation(JsonTypeInfo.class);
+      if (jsonTypeInfo.defaultImpl() != null) {
+        return mapper.constructType(jsonTypeInfo.defaultImpl());
+      }
+    }
+    return primaryType;
+  }
+
+  public static class MappableField {
+    private final List<String> jsonPathToProperty;
+    private final Type propertyType;
+
+    public MappableField(List<String> jsonPathToProperty, Type propertyType) {
+      this.jsonPathToProperty = jsonPathToProperty;
+      this.propertyType = propertyType;
+    }
+
+    public List<String> getJsonPathToProperty() {
+      return jsonPathToProperty;
+    }
+
+    public Type getPropertyType() {
+      return propertyType;
+    }
+
+    public String propertyTypeSimpleName() {
+      if (propertyType instanceof Class) {
+        return ((Class<?>) propertyType).getSimpleName();
+      }
+      return propertyType.getTypeName();
+    }
+
+    public String environmentVariableName() {
+      return jsonPathToProperty.stream().map(String::toUpperCase).collect(Collectors.joining("_"));
```

Comment:
Init value in constructor? The method seems to be invoked several times.

Reply:
fixed